seed/java-sdk/exhaustive/no-custom-config/src/main/java/com/seed/exhaustive/resources/endpoints/container/ContainerClient.java

```diff
@@ -75,6 +83,44 @@ public List<String> getAndReturnListOfPrimitives(List<String> request, RequestOp
         }
     }
 
+    public CompletableFuture<List<String>> getAndReturnListOfPrimitivesAsync(List<String> request) {
+        return getAndReturnListOfPrimitivesAsync(request, null);
+    }
+
+    public CompletableFuture<List<String>> getAndReturnListOfPrimitivesAsync(
+            List<String> request, RequestOptions requestOptions) {
+
+        OkHttpClient client = clientOptions.httpClient();
+        if (requestOptions != null && requestOptions.getTimeout().isPresent()) {
+            client = clientOptions.httpClientWithTimeout(requestOptions);
+        }
+        CompletableFuture<List<String>> future = new CompletableFuture<>();
```

Comment:
if we instantiate a `CompletableFuture` do we need to override the `cancel` method to cancel the underlying okhttp call? we may want a generic utility that extends `CompletableFuture` if this is the case